src/js/personalization/dashboard-beta/containers/DashboardApp.jsx

```diff
@@ -50,10 +52,67 @@ moment.updateLocale('en', {
 });
 
 class DashboardApp extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      modalDismissed: false
+    };
+  }
+
   componentDidMount() {
     scrollToTop();
   }
 
+  dismissModal = () => {
+    this.setState({
+      modalDismissed: true,
+    });
+  }
+
+  renderDowntimeNotification = (status, downtimeWindow, downtimeMap, children) => {
+    switch (status) {
+      case 'downtimeApproaching':
+        return (
+          <div className="downtime-notification row-padded" data-status={status}>
+            <Modal id="downtime-approaching-modal"
+              title="Some parts of your homepage will be down for maintenance soon"
+              status="info"
+              onClose={this.dismissModal}
+              visible={!this.state.modalDismissed}>
+              <p>We’ll be making updates to some tools and features on {downtimeWindow.startTime.format('MMMM Do')} between {downtimeWindow.startTime.format('LT')} and {downtimeWindow.endTime.format('LT')} If you have trouble using parts of the dashboard during that time, please check back soon.</p>
+              <button type="button" className="usa-button-secondary" onClick={this.dismissModal}>Continue</button>
+            </Modal>
+            {children}
+          </div>
+        );
+      default:
+        return children;
+    }
```

Comment:
Ah man, it's kinda unfortunate we have to have so much similar logic as the modal that works by default in DowntimeNotification. The modal content is so customized though that I don't think there would be a good way for the DowntimeNotification API to make it any easier.

Reply:
@ncksllvn this PR is kind of a dirty one to get something up. I think I'd want to explore breaking out the calculation of the Downtime Notification time windows so that it is easier to use a custom notification without having to mess with render functions.

The existing Downtime Notification components can still use those calculation functions, but the expectation would be that if you want to do custom rendering you create whatever container/component you need and just use the calculation functions to get the correct windows and status.

Reply:
That's a great idea. I was originally writing those time windows functions as `static` so we could reuse them, but something, something, and it seemed cleaner not to do it that way in the end. Definitely game to revisit that now we have great use case.